File: estructura_datos/ordenación/bubble_sort/ejemplo.js
const bubbleSort = (a) => {
  const size = a.length;
  let rounds = 0;
  let ord = false;

  while (!ord) {
    ord = true;

    for (let i = 0; i < size - 1 - rounds; i++) {
      if (a[i] > a[i + 1]) {
        let aux = a[i + 1];
        a[i + 1] = a[i];
        a[i] = aux;

        ord = false;
      }
    }

    rounds++;
  }

  return a;
};

console.log(bubbleSort([2, 7, 1, 5, 0, 4, 3, 8, -1, 2, 4]));
console.log(bubbleSort([2, 7, 1, 4, 3, 5, 0, 8, 2, -1, 2]));
console.log(bubbleSort([7, 2, 1, 4, 0, 5, 6, 8, 2]));

File: estructura_datos/ordenación/bubble_sort/ejemplo.py
def bubble_sort(l):
    size = len(l)
    rounds = 0
    ord = False

    while ord == False:
        ord = True

        for i in range(size - 1 - rounds):
            if l[i] > l[i + 1]:
                aux = l[i + 1]
                l[i + 1] = l[i]
                l[i] = aux

                ord = False

        rounds += 1

    return l


def run():
    print(bubble_sort([2, 7, 1, 5, 0, 4, 3, 8, -1, 2, 4]))
    print(bubble_sort([2, 7, 1, 4, 3, 5, 0, 8, 2, -1, 2]))
    print(bubble_sort([7, 2, 1, 4, 0, 5, 6, 8, 2]))


if __name__ == "__main__":
    run()
